### PR TITLE
Propagate access policies in certificates

### PR DESCRIPTION
**Pull request by xacrimon**
This PR allows one to attach access policies locally to a user in addition to dynamically via (future) policy mapping. To allow this, we modify existing certificate and identity code to encode these in relevant places.

This PR is marked do-not-merge, while it is ready for review, it is part of a sequence of staggered PR's based upon eachother that will be merged top-down into dev-predicate when they are all green, which means I'll merge them in a special order.

ptal @fspmarshall @nklaassen

### CODE REVIEW

**Comment by xacrimon** [comment]
@fspmarshall @nklaassen ptal when you can, thanks!

File: lib/auth/auth.go [hunk 1]
```diff
@@ -1394,6 +1394,7 @@ func (a *Server) generateUserCert(req certRequest) (*proto.Certs, error) {
 		Username:          req.user.GetName(),
 		Impersonator:      req.impersonator,
 		Groups:            req.checker.RoleNames(),
+		AccessPolicies:    req.user.GetAccessPolicies(),
```
Comment (nklaassen) [inline lib/auth/auth.go:1397]: this adds them to the tls cert, what about the SSH cert (above)? Would probably be good to add a test that the policies are correctly encoded in both cert types on login

Comment (xacrimon) [reply lib/auth/auth.go:1398]: good catch! I fixed this and added a test.

**Review by fspmarshall — APPROVED** [review]

**Comment by xacrimon** [comment]
@nklaassen can you do another round on this?

**Review by nklaassen — APPROVED** [review]

**Pull request merged**